core/envelope-sim/src/main/kotlin/fr/sncf/osrd/DriverBehaviour.kt

```diff
@@ -0,0 +1,51 @@
+package fr.sncf.osrd
+
+import fr.sncf.osrd.envelope.Envelope
+import fr.sncf.osrd.envelope.MRSPEnvelopeBuilder
+import fr.sncf.osrd.envelope.part.EnvelopePart
+import fr.sncf.osrd.envelope_sim.EnvelopeProfile
+import fr.sncf.osrd.utils.DistanceRangeMap
+import fr.sncf.osrd.utils.distanceRangeMapOf
+import fr.sncf.osrd.utils.units.meters
+import kotlin.math.max
+import kotlin.math.min
+
+data class DriverBehaviour(
+    val acceleratingPostponementOffset: Double = 50.0,
+    val brakingAnticipationOffset: Double = 100.0,
+    val signalingSystems: List<String> = listOf("BAL", "BAPR")
+) {
+    /** Applies the driver behavior to the MRSP, adding reaction time for MRSP changes */
+    fun applyToMRSP(
+        mrsp: Envelope,
+        optSignalingSystemRanges: DistanceRangeMap<String>? = null
```

Comment:
You could just :
```suggestion
        optSignalingSystemRanges: DistanceRangeMap<String>? = distanceRangeMapOf()
```
Would avoid having to put null when unneeded.

Reply:
This is still called from java with v1, where the optional parameters still have to be called explicitly (and IIRC it's more verbose than just `distanceRangeMapOf()` in java)

Reply:
Oh right, true true. I've gotta drop the v1 endpoints x)